File: 1-TwoSum/1-TwoSum.go
// Last updated: 11/7/2025, 2:49:50 PM
func twoSum(nums []int, k int) []int {
    zalupa := make(map[int]int, len(nums))

    for i := range nums {
        value := nums[i]
        if index, ok := zalupa[value]; ok && i != index {
            return []int{index, i}
        }
        zalupa[k-value]=i
    }

    return []int{}
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 11/7/2025, 2:49:51 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp; vector<int> res;
        for(int i = 0; i < nums.size(); i++) {
            if(mp.find(target - nums[i]) != mp.end()) {
                res.push_back(mp[target-nums[i]]);
                res.push_back(i);
                return res;
            }
            mp[nums[i]] = i;
        }
        return res;
    }
};